### PR TITLE
Autotuning

### PR DESCRIPTION
**Pull request by Yanzhiwen001**
Full run on StringTie with more comments

### CODE REVIEW

File: BO_training/bo/optimizer_mixed.py [hunk 1]
```diff
@@ -107,7 +107,12 @@ def restart(self):
                 X_init_cat = self.warp_discrete(X_init_cat, )
             X_init_cat = onehot2ordinal(X_init_cat, self.cat_dims)
             # Put the two parts back by a hstack
-            X_init = np.hstack((X_init_cat, X_init_cont))
+            #X_init = np.hstack((X_init_cat, X_init_cont))
+            #NOTE: change stack manar into insert according to dim of cat
```
Comment (gmarcais) [inline BO_training/bo/optimizer_mixed.py:111]: manar -> manner

File: BO_training/bo/optimizer_mixed.py [hunk 2]
```diff
@@ -154,13 +159,11 @@ def restart(self):
                 )
                 X_init_cat.append(cat_sample)
             X_init_cat = np.array(X_init_cat)
-            #DONE: change stack manar into insert according to dim of cat
+            #NOTE: change stack manar into insert according to dim of cat
```
Comment (gmarcais) [inline BO_training/bo/optimizer_mixed.py:162]: manar -> manner

File: BO_training/bo/optimizer_mixed.py [hunk 3]
```diff
@@ -179,13 +182,11 @@ def restart(self):
...
             # Put the two parts back by a hstack
-            #DONE: change stack manar into insert according to dim of cat 
-            #self.X_init = np.hstack((X_init_cat, X_init_cont))
-            col = 0
-            for i in self.d_cat:
-                new_cont = np.reshape(X_init_cat[:,col], (self.casmopolitan.n_init,1))
-                X_init_cont = np.insert(X_init_cont, [i], new_cont, axis=1)
-                col += 1
+            # self.X_init = np.hstack((X_init_cat, X_init_cont))
+            #NOTE: change stack manar into insert according to dim of cat
```
Comment (gmarcais) [inline BO_training/bo/optimizer_mixed.py:186]: manar -> manner

**Review by gmarcais — APPROVED** [review]
Let's merge this. Looks fine overall.

**Comment by Yanzhiwen001** [comment]
Full run on StringTie with comments

**Pull request merged**